<<<<<<< HEAD
From 35799a3d0b76a85c0b6b6269e02c4418de423e56 Mon Sep 17 00:00:00 2001
=======
From 4ad40a3b4572deee4efc716e7bd6e6e4fda46dcf Mon Sep 17 00:00:00 2001
>>>>>>> 35799a3 (fix truncation error)
From: Sergii Piatakov <sergii.piatakov@globallogic.com>
Date: Thu, 15 Nov 2018 15:28:04 +0200
Subject: [PATCH] fix truncation error

To convert float to integer the truncation is performed, but the
rounding is expected.

Test: Add (4.9, 4.9) should return 10.
Signed-off-by: Sergii Piatakov <sergii.piatakov@globallogic.com>
---
<<<<<<< HEAD
 0001-fix-truncation-error.patch               | 30 +++++++++++
 ...ormatting-use-tabs-instead-of-spaces.patch | 48 ++++++++++++++++++
 0001-improve-calculation-accuracy.patch       | 50 +++++++++++++++++++
 calculator.cpp                                |  4 ++
 4 files changed, 132 insertions(+)
 create mode 100644 0001-fix-truncation-error.patch
 create mode 100644 0001-formatting-use-tabs-instead-of-spaces.patch
 create mode 100644 0001-improve-calculation-accuracy.patch

diff --git a/0001-fix-truncation-error.patch b/0001-fix-truncation-error.patch
new file mode 100644
index 0000000..a60c4c4
--- /dev/null
+++ b/0001-fix-truncation-error.patch
@@ -0,0 +1,30 @@
+From 4ad40a3b4572deee4efc716e7bd6e6e4fda46dcf Mon Sep 17 00:00:00 2001
+From: Sergii Piatakov <sergii.piatakov@globallogic.com>
+Date: Thu, 15 Nov 2018 15:28:04 +0200
+Subject: [PATCH] fix truncation error
+
+To convert float to integer the truncation is performed, but the
+rounding is expected.
+
+Test: Add (4.9, 4.9) should return 10.
+Signed-off-by: Sergii Piatakov <sergii.piatakov@globallogic.com>
+---
+ calculator.cpp | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/calculator.cpp b/calculator.cpp
+index f323e14..d10f529 100644
+--- a/calculator.cpp
++++ b/calculator.cpp
+@@ -2,7 +2,7 @@
+ 
+ int Calculator::Add (double a, double b)
+ {
+-	return a + b;
++	return a + b + 0.5;
+ }
+ 
+ int Calculator::Sub (double a, double b)
+-- 
+2.34.1
+
diff --git a/0001-formatting-use-tabs-instead-of-spaces.patch b/0001-formatting-use-tabs-instead-of-spaces.patch
new file mode 100644
index 0000000..0e9bdd8
--- /dev/null
+++ b/0001-formatting-use-tabs-instead-of-spaces.patch
@@ -0,0 +1,48 @@
+From cbcb06d48af5cb1e0f5673aa7672ca85ea3e2d10 Mon Sep 17 00:00:00 2001
+From: Sergii Piatakov <sergii.piatakov@globallogic.com>
+Date: Thu, 15 Nov 2018 15:26:35 +0200
+Subject: [PATCH 1/2] formatting: use tabs instead of spaces
+
+Signed-off-by: Sergii Piatakov <sergii.piatakov@globallogic.com>
+---
+ calculator.cpp | 4 ++--
+ calculator.h   | 6 +++---
+ 2 files changed, 5 insertions(+), 5 deletions(-)
+
+diff --git a/calculator.cpp b/calculator.cpp
+index b91afea..f323e14 100644
+--- a/calculator.cpp
++++ b/calculator.cpp
+@@ -2,10 +2,10 @@
+ 
+ int Calculator::Add (double a, double b)
+ {
+-    return a + b;
++	return a + b;
+ }
+ 
+ int Calculator::Sub (double a, double b)
+ {
+-    return Add (a, -b);
++	return Add (a, -b);
+ }
+diff --git a/calculator.h b/calculator.h
+index 3740907..d59d596 100644
+--- a/calculator.h
++++ b/calculator.h
+@@ -3,9 +3,9 @@
+ 
+ class Calculator
+ {
+-    public:
+-        int Add (double, double);
+-        int Sub (double, double);
++	public:
++		int Add (double, double);
++		int Sub (double, double);
+ };
+ 
+ #endif//CALCULATOR_H
+-- 
+2.34.1
+
diff --git a/0001-improve-calculation-accuracy.patch b/0001-improve-calculation-accuracy.patch
new file mode 100644
index 0000000..db648c6
--- /dev/null
+++ b/0001-improve-calculation-accuracy.patch
@@ -0,0 +1,50 @@
+From 669f6321a41da2337d975872e9294b8b91139010 Mon Sep 17 00:00:00 2001
+From: Sergii Piatakov <sergii.piatakov@globallogic.com>
+Date: Thu, 15 Nov 2018 15:21:34 +0200
+Subject: [PATCH] improve calculation accuracy
+
+Allow using float point arguments to avoid truncation.
+
+Test: Add (4.9, 4.9) should return 10.
+Signed-off-by: Sergii Piatakov <sergii.piatakov@globallogic.com>
+---
+ calculator.cpp | 4 ++--
+ calculator.h   | 4 ++--
+ 2 files changed, 4 insertions(+), 4 deletions(-)
+
+diff --git a/calculator.cpp b/calculator.cpp
+index 3c63184..b91afea 100644
+--- a/calculator.cpp
++++ b/calculator.cpp
+@@ -1,11 +1,11 @@
+ #include "calculator.h"
+ 
+-int Calculator::Add (int a, int b)
++int Calculator::Add (double a, double b)
+ {
+     return a + b;
+ }
+ 
+-int Calculator::Sub (int a, int b)
++int Calculator::Sub (double a, double b)
+ {
+     return Add (a, -b);
+ }
+diff --git a/calculator.h b/calculator.h
+index 84ff414..3740907 100644
+--- a/calculator.h
++++ b/calculator.h
+@@ -4,8 +4,8 @@
+ class Calculator
+ {
+     public:
+-        int Add (int, int);
+-        int Sub (int, int);
++        int Add (double, double);
++        int Sub (double, double);
+ };
+ 
+ #endif//CALCULATOR_H
+-- 
+2.34.1
+
diff --git a/calculator.cpp b/calculator.cpp
index b91afea..34f8d05 100644
--- a/calculator.cpp
+++ b/calculator.cpp
@@ -2,7 +2,11 @@
 
 int Calculator::Add (double a, double b)
 {
+<<<<<<< HEAD
     return a + b;
+=======
+	return a + b + 0.5;
+>>>>>>> 4ad40a3 (fix truncation error)
=======
 calculator.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/calculator.cpp b/calculator.cpp
index f323e14..d10f529 100644
--- a/calculator.cpp
+++ b/calculator.cpp
@@ -2,7 +2,7 @@
 
 int Calculator::Add (double a, double b)
 {
-	return a + b;
+	return a + b + 0.5;
>>>>>>> 35799a3 (fix truncation error)
 }
 
 int Calculator::Sub (double a, double b)
-- 
2.34.1

